nylas/client/restful_models.py

```diff
@@ -1,17 +1,38 @@
-from datetime import datetime
+from datetime import datetime, date
 
 from nylas.client.restful_model_collection import RestfulModelCollection
 from nylas.client.errors import FileUploadError
 from nylas.utils import timestamp_from_dt
 from six import StringIO
 
+try:
+    from werkzeug.datastructures import MultiDict
+except ImportError:  # pragma: no cover
+    MultiDict = dict
+
 # pylint: disable=attribute-defined-outside-init
 
+def detect_attr(value):
```

Comment:
This function was bothering me a lot because it's named kinda vaguely named even though it does something very specific and we already have `datetime_attrs` lying around. I think the issue here is that we've made `birthday` a weird object that copied aspects from `Event.when`, even though `Event.when` is usually a time-range and not a single date. I've suggested that we change `birthday` to just a normal timestamp, similar to `Thread.first_message_timestamp`. If that goes through, then we can just add `birthday` to the `datetime_attrs` instead.